torch/csrc/jit/export.cpp

```diff
@@ -112,15 +113,59 @@ void addAttribute(onnx::NodeProto * n_p, jit::Node * n, jit::Symbol name) {
   }
 }
 
+void encodeType(onnx::TypeProto* t, Node* n) {
+  onnx::TypeProtoTensorTypeProto* tensor_type = t->mutable_tensor_type();
+  onnx::TypeProtoTensorShapeProto* shape = tensor_type->mutable_shape();
+  JIT_ASSERT(n->hasType());
+  TensorType* node_type = n->type()->expect<TensorType>();
+  const std::vector<std::int64_t>& sizes = node_type->sizes();
+  for (std::int64_t s : sizes) {
+    shape->add_dim(s);
+  }
+  onnx::DataType onnx_type;
+  switch(node_type->scalarType()) {
+    case at::kDouble:
+    case at::kFloat:
+    case at::kHalf:
+      onnx_type = onnx::kFLOAT;
```

Comment:
Really? ONNX has no distinction between fp widths? Are you sure it even supports double and half?

Reply:
https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/export.cpp#L38 This is the original logic. 
But looks like they support double and more types... Let me fix this.
https://github.com/onnx/onnx/blob/master/onnx/onnx.proto#L198

Good catch, thanks.

Reply:
These are just merged yesterday in PR https://github.com/onnx/onnx/pull/57